### ISSUE TITLE
Document TracePropagationTarget for all platforms that have it

### ISSUE THREAD

**Issue by rodolfoBee** [getsentry/sentry-docs#6265]
### Core or SDK?

Platform/SDK

### Which part? Which one?

```[tasklist]
### Tasks
- [ ] .NET SDK
- [x] Java/Android
- [x] Dart/Flutter
```

### Description

Our SDKs automatically add the sentry-trace and baggage headers to all outgoing requests in order to support distributed tracing. In some occasions, users would like to control this to only attach the headers only to certain requests. 

This option is currently not documented for .NET SDK and its frameworks, however the `TracePropagationTarget` does exist in the SDK code:
https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/SentryOptions.cs#L582-L625
https://github.com/getsentry/sentry-dotnet/blob/main/src/Sentry/TracePropagationTarget.cs

### Suggested Solution

Document how it works and add one usage example.

**Comment by chadwhitacre** [comment]
@rbisol Issues only get one `Team: *` assignment. Going with Web SDK here, feel free to adjust.

**Comment by sl0thentr0py** [comment]
this is for @mattjohnsonpint , I'm never sure which team to tag sorry

**Comment by mattjohnsonpint** [comment]
It looks like there are already common shared docs for this, but it's only turned on for `apple` and `node`

https://github.com/getsentry/sentry-docs/blob/9abe7ad5b93b580ecea356f31c46865fd034748c/src/platforms/common/configuration/options.mdx?plain=1#L760-L770

https://docs.sentry.io/platforms/apple/configuration/options/#trace-propagation-targets

https://docs.sentry.io/platforms/node/configuration/options/#trace-propagation-targets


It's also documented in the dev-docs here:
https://develop.sentry.dev/sdk/performance/#tracepropagationtargets

And mentioned as part of Dynamic Sampling for use with the `baggage` header here:
https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#baggage


Thus it's likely a valid option on any SDK that has implemented dynamic sampling.  We should verify and expand the supported platforms, along with the associated PlatformIdentifier for each platform.

In other words - not just .NET

**Comment by kahest** [comment]
React Native: has feature + [docs](https://docs.sentry.io/platforms/react-native/performance/instrumentation/automatic-instrumentation/#tracepropagationtargets), but doesn't use the shared docs block

**Comment by kahest** [comment]
Java/Android: has feature but [docs](https://docs.sentry.io/platforms/android/configuration/options/#tracing-options) miss `tracePropagationTargets`

**Comment by kahest** [comment]
Dart/Flutter: has feature bot no docs

**Comment by marandaneto** [comment]
> Dart/Flutter: has feature bot no docs

https://github.com/getsentry/sentry-docs/pull/6503
Is that all?

**Comment by kahest** [comment]
> > Dart/Flutter: has feature bot no docs
> 
> #6503 Is that all?

Thanks for this! I think we also need to also add .NET and Java.

**Comment by smeubank** [comment]
@bitsandfoxes is this still needed for .NET

**Comment by bitsandfoxes** [comment]
The tasklist is busted. Added the docs for .NET [here](https://github.com/getsentry/sentry-docs/pull/8024).
